File: LogicBuilding/algorithmLogic/common_algorithm.py
arr = [1,2,3,5,6,8,9]
k =3
def sum_max_fixed(arr,k):
    if len(arr) <k:
        return -1
    #sum of first k element
    window_sum = sum(arr[:k])
    max_sum = window_sum
    #sliding window 
    #constatn window
    for i in range (k,len(arr)):
        window_sum += arr[i]-arr[i-k]
        max_sum=max(max_sum,window_sum)
    return max_sum
print(f"The sum of the k elements --> {sum_max_fixed(arr,k)}")

File: LogicBuilding/algorithmLogic/commonAlgorithm.js
const arr = [1,2,3,5,6,7,9]
const k = 3

function maxSumFixed(arr, k) {
  if (arr.length < k) {
    return -1;
  }
  //sum of first k element
  let windowSum = 0;
  for (let i = 0; i < k; i++) {
    windowSum += arr[i];
  }
  //sliding window
  //constant window
  let maxSum = windowSum;
  for (let i = k; i < arr.length; i++) {
    windowSum += arr[i] - arr[i - k];
    maxSum = Math.max(maxSum, windowSum);
  }
  return maxSum;
}
console.log(maxSumFixed(arr,k))
